feat: 팔로우 생성 기능 구현

dain391 opened the pull request:
## 이슈 번호
#17 

## 작업 내용
- **Follow Entity**
  - 팔로워, 팔로이이유 사용자 ID 및 연관관계 설정  

- **DTO 작성**  
  - 팔로우 요청 및 응답에 필요한 DTO 클래스 생성  

- **FollowController**  
  - 팔로우 생성 API 엔드포인트 구현 (POST /follow)  
  - 요청 DTO를 받아 서비스 호출 후 응답 반환  

- **FollowService**  
  - 팔로우 생성 로직 구현  
  - 중복 팔로우 방지 및 기본 예외 처리 예정 (테스트 시 추가 예정)  

- **FollowRepository**  
  - 팔로우 엔티티 저장 및 조회 기능 구현 (JpaRepository 상속) 

## 스크린샷(선택)

dain391 committed:
feat: 팔로우 기능 골격 추가

dain391 committed:
feat: 팔로우 생성 기능 구현

resolves: #17

GyeongSe99 commented on newsfeed/src/main/java/com/npcamp/newsfeed/follow/controller/FollowController.java:
```diff
@@ -0,0 +1,29 @@
+
+    public FollowController(FollowService followService) {
+        this.followService = followService;
+    }
```
코드 통일감을 위해서 생성자를 삭제 하고 @RequiredArgsConstructor 어노테이션을 클래스 위에 추가해주세요

GyeongSe99 commented on newsfeed/src/main/java/com/npcamp/newsfeed/follow/controller/FollowController.java:
```diff
@@ -0,0 +1,29 @@
+    public ResponseEntity<FollowDto> createFollow(@RequestBody FollowDto followDto) {
+        FollowDto createdFollow = followService.createFollow(followDto);
+        return new ResponseEntity<>(createdFollow, HttpStatus.CREATED);
+    }
```
저희 리턴값을 ApiResponse객체로 통일하기로 했는데 해당 부분이 반영이 되지 않은 것 같습니다.
```
    @PostMapping("/signUp")
    public ResponseEntity<ApiResponse<CreateUserResponseDto>> sighUp(@Valid @RequestBody CreateUserRequestDto requestDto) {
        CreateUserResponseDto responseDto = authService.signUp(requestDto.getName(), requestDto.getEmail(), requestDto.getPassword());
        return new ResponseEntity<>(ApiResponse.success(responseDto), HttpStatus.CREATED);
    }
```
위 코드는 AuthController의 태웅님이 작성한 코드인데 
참고하셔서 위와 같이 data를 ApiResponse에 담아서 리턴해주세요

GyeongSe99 commented on newsfeed/src/main/java/com/npcamp/newsfeed/follow/service/FollowService.java:
```diff
@@ -0,0 +1,35 @@
+
+    // 새로운 팔로우 관계 생성
+    @Transactional
+    public FollowDto createFollow(FollowDto followDto) {
```
메소드 재 사용성을 위해서 FollowDto 객체를 전달하는 것이 아니라 안에 있는 필드 하나하나 매개변수로 넘겨주는 방식으로 변경해주세요.

GyeongSe99 commented on newsfeed/src/main/java/com/npcamp/newsfeed/follow/service/FollowService.java:
```diff
@@ -0,0 +1,35 @@
+                .followerUserId(saved.getFollowerUserId())
+                .followeeUserId(saved.getFolloweeUserId())
+                .build();
+    }
```
entity -> dto 변환 시 Dto클래스 하위에 toDto라는 이름으로 만든 메소드를 사용해서 FollowDto객체 만들어주세요

dain391 committed:
refactor: 컨트롤러 및 서비스 클래스에 컨벤션 적용

- 생성자 제거
- 요청/응답 DTO 분리
- 서비스 메서드 파라미터 방식 변경
- DTO 변환 시 fromEntity() 사용
- API 응답을 ApiResponse로 통일

resolves: #17

GyeongSe99 commented on newsfeed/src/main/java/com/npcamp/newsfeed/follow/dto/FollowResponseDto.java:
```diff
@@ -0,0 +1,20 @@
+    private Long followerUserId;
+    private Long followeeUserId;
+
+    public static FollowResponseDto fromEntity(Follow follow) {
+        return FollowResponseDto.builder()
+                .followerUserId(follow.getFollowerUserId())
+                .followeeUserId(follow.getFolloweeUserId())
+                .build();
+    }
```
메소드명 toDto로 통일해주세요!

dain391 committed:
style: 메소드명 fromEntity를 toDto로 변경하여 DTO 변환 컨벤션 통일

- FollowResponseDto 클래스 내 메소드명 수정
- 서비스 및 컨트롤러에서 해당 메소드명 변경 반영

resolves: #17
